src/web3/WalletClient.ts

```diff
@@ -510,7 +489,7 @@ export class WalletClient extends BaseClient implements IWalletClient {
     }
 
     // convert signature to base58
-    const version = Buffer.from(varintEncode(VERSION_NUMBER));
+    const version = Buffer.from(varintEncode(VERSION_NUMBER)); // issue here
```

Comment:
Will be replaced by: `const version = Buffer.from(varintEncode(secretKey.version));`
Signature version will be secretKey version, or we should put sig version in arg of the function

Reply:
@AurelienFT